pkg/adaptor/proxy/proxy.go

```diff
@@ -64,12 +66,18 @@ func (p *agentProxy) Start(ctx context.Context, serverURL *url.URL) error {
 
 	close(p.readyCh)
 
-	proxyService := newProxyService()
+	proxyService := newProxyService(p.criSocketPath)
 
 	if err := proxyService.connect(ctx, serverURL.Host); err != nil {
 		return err
 	}
 
+	err = proxyService.initCriClient(ctx)
+	if err != nil {
+		// cri client is optional currently, we ignore any errors here
+		logger.Printf("failed to init cri client, the err: %v", err)
+	}
+
```

Comment:
Instead of calling `initCriClient` from `agentProxy.Start`, we can directly initialize a GRPC client in `agentProxy.Start` using `criSocketPath`, and pass the client to `newProxyService` as its argument.

Reply:
Agreed.

Reply:
@yoheiueda comments addressed, thanks for the review and very good suggestions!